### PR TITLE
MM-30077: Prevent error during LDAP sync when there's replication lag.

### PR DESCRIPTION
**Pull request by mkraft**
#### Summary

* Adds a new app method to get team member record specifically from master, not a read replica.
* Adds a parameter to the team store to read a member from the master database.
* Switches to the new app method in a few cases to unblock LDAP sync when it writes a team member and then was trying to immediately read it from a replica.

#### Ticket Link

https://mattermost.atlassian.net/browse/MM-30077

#### Release Note

```release-note
Added a new app method GetTeamMemberFromMaster. Reads team members explicitly from the master database when adding a user to a channel using the AddUserToChannel app method and when adding a users to channels via group sync.
```

### CODE REVIEW

File: app/channel.go [hunk 1]
```diff
@@ -1370,7 +1370,7 @@ func (a *App) addUserToChannel(user *model.User, channel *model.Channel, teamMem
 }
 
 func (a *App) AddUserToChannel(user *model.User, channel *model.Channel) (*model.ChannelMember, *model.AppError) {
-	teamMember, nErr := a.Srv().Store.Team().GetMember(channel.TeamId, user.Id)
+	teamMember, nErr := a.Srv().Store.Team().GetMember(channel.TeamId, user.Id, true)
```
Comment (mkraft) [inline app/channel.go:1373]: This change is not localized to just LDAP sync, but I'm assuming the performance impact will be low.

**Comment by agnivade** [comment]
@mkraft - I thought we wanted to use the new context mechanism introduced for this?

**Comment by mkraft** [comment]
> @mkraft - I thought we wanted to use the new context mechanism introduced for this?

Yes you're right. I totally forgot you had added that.

**Review by reflog — APPROVED** [review]
code-wise looks good, but should probably be load-tested in cluster environment to assess the impact

**Comment by amyblais** [comment]
@mkraft Is this still on track to have this ready for QA by the end of this week?

**Comment by mkraft** [comment]
> @mkraft Is this still on track to have this ready for QA by the end of this week?

This PR could be ready but it turns out it doesn't fully satisfy the needs of Dancing Penguin as-is. I've got some other working in progress to add. When it's ready we'll have to review whether it's appropriate for a dot release.

**Comment by mkraft** [comment]
Closing this in favour of https://github.com/mattermost/mattermost-server/pull/16560.

**Pull request closed without merging**